packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -0,0 +1,351 @@
+    return new Proxy(obj, {
+        get: (target: T, property: string) => {
+            const override = overrides[property as keyof T];
+            if (override) {
+                if (typeof override === "function") {
+                    return override(target);
+                }
+                const real = target[property as keyof T];
+                if (typeof real === "function") {
+                    return (...args: any) => {
+                        const res = real.bind(target)(...args);
+                        if (res.then !== undefined) {
+                            return res.then((v) => createFunctionOverrideProxy(v, override));
+                        }
+
+                        return createFunctionOverrideProxy(res, override);
+                    };
+                }
+
+                return createFunctionOverrideProxy(real as any, override);
+            }
+
+            return target[property];
+        },
+    });
+}
+
+async function runAndValidateBatch(
+    provider: ITestObjectProvider,
+    proxyDsf: IDocumentServiceFactory,
+    timeout: number,
+    ) {
+    let containerUrl: string | undefined;
+    {
+        const loader = provider.createLoader(
+            [[
+                provider.defaultCodeDetails,
+                provider.createFluidEntryPoint(),
+            ]]);
+
+        const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+        await container.attach(provider.driver.createCreateNewRequest(Date.now().toString()));
+        containerUrl = await container.getAbsoluteUrl("");
+        container.close();
+    }
+    assert(containerUrl);
+    {
+        const loader = provider.createLoader(
+            [[
+                provider.defaultCodeDetails,
+                provider.createFluidEntryPoint({ runtimeOptions: { summaryOptions: { disableSummaries: true } } }),
+            ]],
+            {
+                documentServiceFactory: proxyDsf,
+            });
+        const container = await loader.resolve({ url: containerUrl });
+        const testObject = await requestFluidObject<TestFluidObject>(container, "default");
+        // send batch
+        testObject.context.containerRuntime.orderSequentially(() => {
+            for (let i = 0; i < 10; i++) {
+                testObject.root.set(i.toString(), i);
+            }
+        });
+        // send non-batch
+        testObject.root.set("foo", "bar");
+        while (container.isDirty && !container.closed) {
+            await timeoutPromise((resolve, reject) => {
+                container.once("saved", () => resolve());
+                container.once("closed", (e) => reject(e));
+            },
+            {
+                durationMs: timeout, // 60 * 60 * 1000,
+            });
+        }
+
+        for (let i = 0; i < 10; i++) {
+            assert.equal(testObject.root.get(i.toString()), i, i.toString());
+        }
+        assert.equal(testObject.root.get("foo"), "bar");
```

Comment:
Can you please add messages to assert so its slightly easier to debug when something fails in CI.

Reply:
```suggestion
        assert.equal(testObject.root.get("foo"), "bar", "validate after batch op");
```